modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -621,9 +625,27 @@ protected void doInitialise() throws InitialisationException {
     flowConstruct = new LazyValue<>(() -> (FlowConstruct) componentLocator.find(getRootContainerLocation()).orElse(null));
     messageProcessContext = createProcessingContext();
     if (shouldRunOnThisNode()) {
+      if (LOGGER.isDebugEnabled()) {
+        boolean isPrimaryPollingInstance = clusterService.isPrimaryPollingInstance();
+        if (primaryNodeOnly) {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, it is running on the primary node and it must only run on this node.",
+                     sourceModel.getName(), getLocation().getRootContainerName());
+        } else {
+          LOGGER
+              .debug("Message source '{}' on flow '{}' is being initialised, there are no cluster restrictions regarding which node must run the source. This node {} the primary node.",
+                     sourceModel.getName(), getLocation().getRootContainerName(), isPrimaryPollingInstance ? "is" : "is not");
+        }
+      }
+
       reallyDoInitialise();
     } else {
+      LOGGER
+          .debug("Message source '{}' on flow '{}' is not being initialised because it should only run on the primary node of the cluster.",
+                 sourceModel.getName(), getLocation().getRootContainerName());
       new PrimaryNodeLifecycleNotificationListener(() -> {
+        LOGGER.debug("Message source '{}' on flow '{}' is being initialised because it became the primary node of the cluster.",
```

Comment:
Message source '{}' on flow '{}' became primary node of the cluster and is initializing.

Reply:
My wording was wrong in the first place. What became the primary node is not the source, what about:

Message source '{}' on flow '{}' is initializing because the node became the primary node of the cluster.

Reply:
`node became the primary node of the cluster.` -> `the node became the cluster's primary`

Reply:
Message source '{}' on flow '{}' is initializing because the node became cluster's primary.